src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -66,6 +66,11 @@ public partial class DbOnTheRocks : IDb, ITunableDb, IReadOnlyNativeKeyValueStor
 
     private readonly DbSettings _settings;
 
+    // ReSharper disable once CollectionNeverQueried.Local
+    // Need to keep options from GC in case of merge operator applied, as they are used in callback
+    // TODO: find better way?
+    private readonly ConcurrentBag<OptionsHandle> _doNotGcOptions = [];
```

Comment:
Open for discussion.

Need to preserve created `OptionsHandle` from GC as they are used for native merge callbacks.

Reply:
ConcurrentBag is very peculiar collection and something you probably don't want. I think simple List with locks is better here. Otherwise ConcurrentSet.